R/strip_html.r

```diff
@@ -0,0 +1,45 @@
+#' @title strip_html
+#'
+#' @description
+#' Remove all values between "<>", which are typically HTML tags.
+#'
+#' @param dat a data.frame.
+#' @param ignore a character vector containing values to ignore while stripping HTML tags. For instance, if you have <keep me> and <me too> in your column names, add ignore = c("keep me", "me too").
+#'
+#'
+#' @return a data.frame object.
+#'
+#' @export
+strip_html <- function(dat, ignore = NULL){
+
+  regex.escape <- function(string) {
+    gsub("([][{}()+*^$|\\\\?.])", "\\\\\\1", string)
+  }
+
+
+
+  if (is.null(ignore)){
+    check_ignore <- ""
+    names(dat) <- gsub("(<[^>]*>)","",names(dat))
+  }
+
+  if (!is.null(ignore)){
+   check_ignore <- paste(ignore, collapse = "|")
+   names(dat) <- gsub(paste0("<(?!(?:", paste(regex.escape(ignore), collapse="|"), ")>)[^>]*>"), "", names(dat), perl=TRUE)
+  }
+
+  if (!is.null(ignore) & any(grepl(check_ignore, names(dat))) == F){
```

Comment:
Should this be comparing `ignore`, not `check_ignore`?  I think if `check_ignore` is collapsed with `|` it won't have any matches?  Maybe I'll see I'm wrong when I look at tests.

Reply:
Maybe I'm having trouble here b/c it feels odd to me to compare to the post-corrected names of `dat`.  But if it all works and I'm just not following correctly as I glance through, let me know and I'll just accept being wrong 😁

Reply:
There's almost definitely a better way of doing this, but my logic is: 

- Accept a character vector for `ignore` argument
- `grepl` argument `pattern=` accepts an argument of length = 1
- So, convert character vector to a character vector with an OR operator
- If `grepl` doesn't find any matches, then a warning shows up that none of the supplied values to ignore were present in the column names.

Reply:
Ah the `|` operator make sense, I was mixed up about `grepl`.

What do you think about moving the check `if (!is.null(ignore) & any(grepl(check_ignore, names(dat))) == F){` to before `names(dat)` are stripped?  I don't think it should matter if the code all works as it should, but it feels more logical to me to check for the presence of the ignore strings before ignoring them, rather than checking the result.  If it's harder to match `<ignore this>` than `ignore this` than my suggestion isn't worth it.

Reply:
Great point! I moved this code up so that it's the first check if `!is.null(ignore)`.